### PR TITLE
Fix playback speed controls on small screens

### PR DESCRIPTION
**Pull request by jameswklassen**
## Issue
On smaller screens, the playback controls popup would overflow the video, causing the top of the list to be cut off

## Solution
Add scrolling overflow to the menu when the height of the video is too small.
This is achieved with a resize event listener which checks the height of the element and sizes it accordingly.

### Before - menu overflowing, top elements inaccessible
overflowing

### After - scrollable menu
fixed

### CODE REVIEW

File: d2l-video.js [hunk 1]
```diff
@@ -320,6 +327,7 @@ Polymer({
 	behaviors: [
 		window.D2L.MediaBehavior,
 		IronA11yKeysBehavior,
+		IronResizableBehavior,
```
Comment (apalaniuk) [inline d2l-video.js:330]: I think we could also use web APIs like [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver), but given this buys heavily into Polymer behaviours already, I'm fine with using this for now.

File: d2l-video.js [hunk 2]
```diff
@@ -379,6 +386,13 @@ Polymer({
 		'esc': '_closeControls'
 	},
 
+	ready: function() {
```
Comment (jstefaniuk-d2l) [inline d2l-video.js:389]: we should make sure to unobserve height when this element is no longer around (often these are done on attached/detached)

File: d2l-video.js [hunk 3]
```diff
@@ -498,13 +512,25 @@ Polymer({
...
 		this.shadowRoot.querySelector('#controlBar').removeAttribute('hidden');
+		this._handleControlHeight();
 	},
 
 	_hideControls: function() {
 		this._controlsVisible = false;
 		this.shadowRoot.querySelector('#controlBar').setAttribute('hidden', '');
 	},
 
+	_handleControlHeight: function() {
```
Comment (jstefaniuk-d2l) [inline d2l-video.js:523]: you may be able to do this in one line: `control.style.maxHeight = `${this.offsetHeight - 60}px`;` (if you've set the height value to the desired normal height)

Also I won't send you too far down the path, but you could potentially do it all with css if you had `position: relative` on the top level and then used the `max-height` property on the control, as long as nothing between had the `position: relative` property

Comment (jameswklassen) [reply d2l-video.js:523]: Oh yeah, good call on the max height. Interesting about doing it with pure CSS though, might look into that.

**Comment by jameswklassen** [comment]
I spent some time trying to solve this with vanilla CSS but I wasn't able to get something working with the current structure. 😞

File: package.json [hunk 4]
```diff
@@ -47,7 +48,7 @@
     "@d2l/seek-bar": "Brightspace/d2l-seek-bar#semver:^1",
     "@polymer/iron-a11y-keys-behavior": "^3.0.0-pre.18",
     "@polymer/iron-flex-layout": "^3.0.0-pre.18",
-    "@polymer/iron-range-behavior": "^3.0.0-pre.18",
+    "@polymer/iron-resizable-behavior": "^3.0.1",
```
Comment (apalaniuk) [inline package.json:51]: Looks like we can revert this change now

Comment (jameswklassen) [reply package.json:51]: 🤦

File: package.json [hunk 5]
```diff
@@ -47,7 +48,6 @@
     "@d2l/seek-bar": "Brightspace/d2l-seek-bar#semver:^1",
     "@polymer/iron-a11y-keys-behavior": "^3.0.0-pre.18",
     "@polymer/iron-flex-layout": "^3.0.0-pre.18",
-    "@polymer/iron-range-behavior": "^3.0.0-pre.18",
```
Comment (jameswklassen) [inline package.json:50]: It was inadvertently discovered that this dependency wasn't needed.

To test I started fresh and removed the dependency. I tested with a screen reader, as well as in Chrome, Credge, Firefox, IE11.
